core/email.go

```diff
@@ -16,34 +16,38 @@ var (
 	ErrEmailAddressEmpty   = errors.New("e-mail address is empty")
 )
 
-type EmailAddress interface {
-	// String returns the string representation for this e-mail address
-	String() string
-}
-
-type emailAddress struct {
+type EmailAddress struct {
 	address string
 }
 
-func (email emailAddress) String() string {
+func (email *EmailAddress) String() string {
 	return email.address
 }
 
+func (email *EmailAddress) UnmarshalText(text []byte) error {
+	add, err := NewEmailAddress(string(text))
+	if err != nil {
+		return err
+	}
+	email.address = add.address
+	return nil
+}
+
 // NewEmailAddress parses an e-mail address from any string.
 // This uses RFC-5322 to determine valid e-mail addresses, e.g. "Biggie Smalls <notorious@example.com>"
 func NewEmailAddress(address string) (EmailAddress, error) {
```

Comment:
Anders moeten we hier een pointer teruggeven? Dan kan nil ook. In wachtplanner komen beiden voor. Ik hou momenteel niet echt zo van dat lege emailadres, maar er zijn heel wat plaatsen waar die interface nil mocht zijn, maar waar het geen pointer was. Er zijn zelfs heel wat tests die nu in mijn ogen wat overbodig zijn, want een input met leeg EmailAddress zou eigenlijk nooit mogen kunnen voorkomen - die tests met nil waren daarom in mijn ogen ook al niet ideaal. Bv voor CreateUser: als het een EmailAddress heeft als input, zou ik aannemen dat die email geldig is.

Reply:
Ja, dit zou gerust een pointer mogen teruggeven. Misschien moeten we het dan ook "ParseEmailAddress" maken ipv New om wat consistenter te zijn? 😛